refector the xpt2046 so it will be a platform for the touchscreen component

nielsnl68 opened the pull request:
** I Closed this PR because i messed up my last try to get my source up-to-date **

See the new PR here: PR#3793

----------------------------------------------------------------------

- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

# What does this implement/fix? 
it will now use the Touchscreen eco system as a platform component.

Quick description and explanation of changes
Before the xpt2046 could only be used as a standalone componet, missing all the futures that the Touchscreen platform has,
this is now being fixed.

The old method is fully removed. howhever there should be a error warning to expain that you should use it as platform of the touchscreen.


## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Other: 
All existing yaml keys are still vaild and for the binary_sensors the platform need to be updated to `touchscreen` instead of `xtp2046`


**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2259

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
touchscreen:
  - platform: xpt2046
    display: my_display
    id: my_toucher
    cs_pin: 14
    #interupt_pin: 16
    report_interval: 50ms
    threshold: 400
    calibration_x_min: 3860
    calibration_x_max: 280
    calibration_y_min: 340
    calibration_y_max: 3860
    #swap_x_y: true
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

nielsnl68 committed:
refector the xpt2046 so it will be a platform
of the touchscreen component
- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

nielsnl68 committed:
refector the xpt2046 so it will be a platform
of the touchscreen component
- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

nielsnl68 committed:
refector the xpt2046 so it will be a platform
of the touchscreen component
- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

nielsnl68 committed:
refector the xpt2046 so it will be a platform
of the touchscreen component
- removing his own binarysensor component.
- when interupt pin is set it will now react on the event.
- otherwise it will start polling as the old behavier dit.

nielsnl68 reviewed (commented):
not sure what this will do,

nielsnl68 committed:
Update the test file with the new config

nielsnl68 committed:
Update the test file with the new config

nielsnl68 committed:
Update the test file with the new config

nielsnl68 committed:
Update most error reported by the PR

nielsnl68 committed:
Update most error reported by the PR

nielsnl68 committed:
Update most error reported by the PR

nielsnl68 committed:
Update most error reported by the PR

nielsnl68 committed:
other warning i overlooked

nielsnl68 committed:
other warning i overlooked

nielsnl68 committed:
other warning i overlooked

nielsnl68 committed:
someother fixes i forgot.

nielsnl68 committed:
someother fixes i forgot.

nielsnl68 committed:
someother fixes i forgot.

nielsnl68 committed:
fix error

nielsnl68 committed:
fix error

nielsnl68 committed:
fix error

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

nielsnl68 committed:
lastest fix

numo68 commented:
Hello @nielsnl68 , thanks for porting this to the touchscreen platform! - frankly I did not follow that part of ESPHome so I did not even know it is now possible. Unfortunately I won't be able to test with my setup used to develop the original xpt2046 in the near future but from a casual review the PR looks fine to me.

nielsnl68 committed:
Finished the touch position calculation
- the has just a 12bit ADC normalistion to 0x7fff is overkil
- using the display width, height and rotation now directly.
- removed old properties from the class

nielsnl68 committed:
Finished the touch position calculation
- the has just a 12bit ADC normalistion to 0x7fff is overkil
- using the display width, height and rotation now directly.
- removed old properties from the class

nielsnl68 committed:
Finished the touch position calculation
- the has just a 12bit ADC normalistion to 0x7fff is overkil
- using the display width, height and rotation now directly.
- removed old properties from the class

nielsnl68 committed:
Finished the touch position calculation
- the has just a 12bit ADC normalistion to 0x7fff is overkil
- using the display width, height and rotation now directly.
- removed old properties from the class

nielsnl68 committed:
fixed a couple of small things

nielsnl68 committed:
fixed a couple of small things

nielsnl68 committed:
fixed a couple of small things

nielsnl68 committed:
fixed a couple of small things

nielsnl68 committed:
clang fixes

nielsnl68 committed:
clang fixes

nielsnl68 committed:
clang fixes

nielsnl68 committed:
clang fixes

nielsnl68 commented:
> 

Did you ever did some test with the interrupt used?  I changed it a bit so it is now using the esphomes interrupt mechanic. But i was not able to test it.
Not sure of the pin will stay active as long you press the screen and move around or that it will blink per movement. (i hope the last thing)

jesserockz commented on esphome/components/xpt2046/__init__.py:
```diff
@@ -1,129 +0,0 @@
-import esphome.codegen as cg
```
Make sure to leave a basic `CONFIG_SCHEMA` here with `cv.invalid` letting users know to check the docs. Look for other examples in the codebase

nielsnl68 replied:
Good idea, i added it also for the old `binary_sensor` and the remove/changed keys.

nielsnl68 committed:
add warning for configuration using the old method

nielsnl68 committed:
add warning for configuration using the old method

nielsnl68 committed:
add warning for configuration using the old method

nielsnl68 committed:
Merge branch 'dev' into nvds-xpt2046-as-platform

nielsnl68 committed:
I found some issue in the ili9341 driver (#3756)

nielsnl68 committed:
Merge branch 'nvds-xpt2046-as-platform' of https://github.com/nielsnl68/esphome into nvds-xpt2046-as-platform

anatoly-savchenkov committed:
Ignore NaN states in the integration component (#3767)

Commit:
Add the same docker tags as used in HA (#3752)

dependabot[bot] committed:
Bump pytest from 7.1.1 to 7.1.3 (#3766)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Avirsaam committed:
Update modbus_controller.cpp (#3768)

jesserockz committed:
Remove unneeded line (spi component adds it) (#3778)

jesserockz committed:
Move crc16 to helpers (#3780)

jesserockz committed:
YAML linting (#3779)

jesserockz committed:
Fix HA addon auth using HA credentials (#3758)

Philippe12 committed:
Add support to tm1621 display (#3737)

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

pvizeli committed:
u-fire EC sensor (#3774)

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

kbickar committed:
Add support for MPL3115A2 Pressure/Altitude and Temperature Sensor (#3371)

* Add support for mpl3115a2

* Add codeowner

* Linter/test updates

* Minor changes

* Made pressure/altitude exclusive

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

dbuezas committed:
Add support for BL0942 voltage, current, energy and power Sensor (#3777)

dependabot[bot] committed:
Bump pylint from 2.15.0 to 2.15.2 (#3785)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

anatoly-savchenkov committed:
Add Factory Reset button and switch (#3724)

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

CarlosGS committed:
[MPU6050] Support devices with WHOAMI 0x98 (#3784)

dependabot[bot] committed:
Bump zeroconf from 0.39.0 to 0.39.1 (#3782)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nielsnl68 committed:
Merge branch 'nvds-xpt2046-as-platform' of https://github.com/nielsnl68/esphome into nvds-xpt2046-as-platform

nielsnl68 committed:
remove unneeded log message

nielsnl68 committed:
removed resetting display settings
in favor of reading them athock
until there is a a way that components
can tell others to update there local settings

nielsnl68 committed:
fixed clang format issues

nielsnl68 committed:
Merge branch 'nvds-xpt2046-as-platform' of https://github.com/nielsnl68/esphome into nvds-xpt2046-as-platform

nielsnl68 committed:
Merge branch 'nvds-xpt2046-as-platform' of https://github.com/nielsnl68/esphome into nvds-xpt2046-as-platform

nielsnl68 committed:
hope this will help
